lib/views/setup.js

```diff
@@ -34,155 +67,125 @@ module.exports = function (req, res) {
-  config.set('databasePrefix', settings.instanceUrl)
-  config.set('themeParameters', { 'default': { baseColor: settings.color } })
-  config.set('frontPageText', settings.frontPageText)
-  config.set('allowPublicSignup', settings.allowPublicSignup)
-  config.set('databasePrefix', settings.uriPrefix)
-
-  config.set('triplestore', extend(config.get('triplestore'), {
-    graphPrefix: settings.uriPrefix,
-    defaultGraph: settings.uriPrefix + 'public',
-    virtuosoINI: settings.virtuosoINI,
-    virtuosoDB: settings.virtuosoDB
-  }))
+  config.set('instanceUrl', updatedSettings.instanceUrl)
+  config.set('webOfRegistries', { [updatedSettings.uriPrefix.slice(0, -1)]: updatedSettings.instanceUrl.slice(0, -1) })
+  config.set('databasePrefix', updatedSettings.instanceUrl)
+  config.set('themeParameters', { 'default': { baseColor: updatedSettings.color } })
+  config.set('frontPageText', updatedSettings.frontPageText)
+  config.set('allowPublicSignup', updatedSettings.allowPublicSignup)
+  config.set('databasePrefix', updatedSettings.uriPrefix)
+
+  config.set('triplestore', {
+    ...config.get('triplestore'),
+    graphPrefix: updatedSettings.uriPrefix,
+    defaultGraph: updatedSettings.uriPrefix + 'public',
+    virtuosoINI: updatedSettings.virtuosoINI,
+    virtuosoDB: updatedSettings.virtuosoDB
+  })
 
-  createUser({
+  config.set('externalAuth', {
+    enabled: !!updatedSettings.authProvider,
+    provider: updatedSettings.authProvider || null,
+    google: {
+      clientId: updatedSettings.googleClientId,
+      clientSecret: updatedSettings.googleClientSecret
+    }
+  })
 
-    username: settings.userName,
-    name: settings.userFullName,
-    email: settings.userEmail,
-    affiliation: settings.affiliation || '',
-    password: settings.userPassword,
-    isAdmin: true,
-    isCurator: true,
-    isMember: true
+  try {
+    const user = await createUser({
+      username: updatedSettings.userName,
+      name: updatedSettings.userFullName,
+      email: updatedSettings.userEmail,
+      affiliation: updatedSettings.affiliation || '',
+      password: updatedSettings.userPassword,
+      isAdmin: true,
+      isCurator: true,
+      isMember: true
+    })
 
-  }).then((user) => {
     config.set('firstLaunch', false)
 
-    req.session.user = user.id
+    await theme.setCurrentThemeFromConfig()
+    await getRegistries()
 
-    return theme.setCurrentThemeFromConfig()
-  }).then(() => {
-    return getRegistries()
-  }).then(() => {
+    req.session.user = user.id
     req.session.save(() => {
       res.redirect(req.body.next || '/')
     })
-  }).catch((err) => {
-    errors.push('Could not create user')
-
-    errors.push(err.stack)
-
-    return setupForm(req, res, settings, {
-      errors: errors
+  } catch (err) {
+    return setupForm(req, res, updatedSettings, {
+      errors: [
+        'Could not create user',
+        err.stack
```

Comment:
Should the stack be spread?

Reply:
No. Prior to this changes to error was pushed using following code:

```js
     errors.push('Could not create user')
     errors.push(err.stack) 
```